pkg/mon/logger_output_iowriter.go

```diff
@@ -0,0 +1,38 @@
+package mon
+
+import (
+	"fmt"
+	"github.com/applike/gosoline/pkg/clock"
+	"io"
+	"os"
+)
+
+func newIowriterLoggerWithInterfaces(clock clock.Clock, formatter formatter, output *os.File, timestampFormt string) (Handler, error) {
+	return &iowriterLogger{
+		clock:           clock,
+		formatter:       formatter,
+		output:          output,
+		timestampFormat: timestampFormt,
+	}, nil
+}
+
+type iowriterLogger struct {
+	clock           clock.Clock
+	formatter       formatter
+	output          io.Writer
+	timestampFormat string
+}
+
+func (s *iowriterLogger) Write(level string, msg string, logErr error, metadata Metadata) {
+	timestamp := s.clock.Now().Format(s.timestampFormat)
+
+	buffer, err := s.formatter(timestamp, level, msg, logErr, &metadata)
+	if err != nil {
+		_, _ = fmt.Fprintf(os.Stderr, "Failed to format log, %v\n", err)
+	}
+
+	_, err = s.output.Write(buffer)
```

Comment:
What happens if only part of the log message got written? I know we didn't in the past, but maybe you should check the number of written bytes, especially should the io.Writer be something like a network stream

Reply:
and then log the error to... what? ^^

Reply:
The same place you are logging the error you can already get (stderr that is). But as long as the writer abides to the rules of `io.Writer`, we should always get an error for a partial write (that is just completely different from the libc write which might return success for a partial write...)